core/lib/multivm/src/utils.rs

```diff
@@ -0,0 +1,164 @@
+            crate::vm_m5::transaction_data::derive_overhead(
+                gas_limit,
+                gas_price_per_pubdata,
+                encoded_len,
+            )
+        }
+        VmVersion::M6Initial | VmVersion::M6BugWithCompressionFixed => {
+            crate::vm_m6::transaction_data::derive_overhead(
+                gas_limit,
+                gas_price_per_pubdata,
+                encoded_len,
+                crate::vm_m6::transaction_data::OverheadCoefficients::from_tx_type(tx_type),
+            )
+        }
+        VmVersion::Vm1_3_2 => crate::vm_1_3_2::transaction_data::derive_overhead(
+            gas_limit,
+            gas_price_per_pubdata,
+            encoded_len,
+            crate::vm_1_3_2::transaction_data::OverheadCoefficients::from_tx_type(tx_type),
+        ),
+        VmVersion::VmVirtualBlocks => crate::vm_virtual_blocks::utils::overhead::derive_overhead(
+            gas_limit,
+            gas_price_per_pubdata,
+            encoded_len,
+            crate::vm_virtual_blocks::utils::overhead::OverheadCoefficients::from_tx_type(tx_type),
+        ),
+        VmVersion::VmVirtualBlocksRefundsEnhancement => {
+            crate::vm_refunds_enhancement::utils::overhead::derive_overhead(
+                gas_limit,
+                gas_price_per_pubdata,
+                encoded_len,
+                crate::vm_refunds_enhancement::utils::overhead::OverheadCoefficients::from_tx_type(
+                    tx_type,
+                ),
+            )
+        }
+        VmVersion::VmBoojumIntegration => {
+            crate::vm_boojum_integration::utils::overhead::derive_overhead(
+                gas_limit,
+                gas_price_per_pubdata,
+                encoded_len,
+                crate::vm_boojum_integration::utils::overhead::OverheadCoefficients::from_tx_type(
+                    tx_type,
+                ),
+            )
+        }
+    }
+}
+
+pub fn get_bootloader_encoding_space(version: VmVersion) -> u32 {
+    match version {
+        VmVersion::M5WithRefunds | VmVersion::M5WithoutRefunds => {
+            crate::vm_m5::vm_with_bootloader::BOOTLOADER_TX_ENCODING_SPACE
+        }
+        VmVersion::M6Initial | VmVersion::M6BugWithCompressionFixed => {
+            crate::vm_m6::vm_with_bootloader::BOOTLOADER_TX_ENCODING_SPACE
+        }
+        VmVersion::Vm1_3_2 => crate::vm_1_3_2::vm_with_bootloader::BOOTLOADER_TX_ENCODING_SPACE,
+        VmVersion::VmVirtualBlocks => {
+            crate::vm_virtual_blocks::constants::BOOTLOADER_TX_ENCODING_SPACE
+        }
+        VmVersion::VmVirtualBlocksRefundsEnhancement => {
+            crate::vm_refunds_enhancement::constants::BOOTLOADER_TX_ENCODING_SPACE
+        }
+        VmVersion::VmBoojumIntegration => {
+            crate::vm_boojum_integration::constants::BOOTLOADER_TX_ENCODING_SPACE
+        }
+    }
+}
+
+pub fn get_max_transactions_in_batch(version: VmVersion) -> usize {
```

Comment:
maybe specify that this has to do with the bootloader? Something like `get_bootloader_l1_batch_tx_limit`